### ISSUE TITLE
Add multiple tag search to unified search

### ISSUE THREAD

**Issue by Discostu36** [nextcloud/server#37302]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
I organize my files by tags. It is possible to use the unified search to find files that have a certain tag. I can, for example, search for "receipt" and get all files with the tag `receipt`. But if I want to search for files tagged with `receipt `and `clothes`, I can't use the unified search. Entering "receipt clothes" doesn't work. I have to go to `https://nextcloud.example.com/apps/files/?dir=/&view=systemtagsfilter` and enter them there to do that.

**Describe the solution you'd like**
Add the possibility to search for multiple tags to the unified search. I am not sure yet how this could best be done, as, as far as I know, unified search currently doesn't use space as a delimiter of search terms but always matches to the whole string. But maybe this paradigm could be changed? Alternatively, maybe auto-suggest for tags could be added?

**Describe alternatives you've considered**
None

**Additional context**
None.

**Comment by joshtrichards** [comment]
Duplicate of #4851

Please subscribe to that issue to monitor progress and give it a thumbs up to express your interest.

**Comment by Discostu36** [comment]
I disagree that this is a duplicate as this ticket is about unified search and the other one is not.
